product_replenishment_cost/models/product_template.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+# © 2012 Alexandre Fayolle,Yannick Vaucher,Joël Grand-Guillaume,Camptocamp
+# © 2016 Sodexis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, fields, api
+import openerp.addons.decimal_precision as dp
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    @api.multi
+    @api.depends('standard_price')
+    def _get_replenishment_cost(self):
+        replenishment_cost = 0.0
```

Comment:
If a template has no variant, it will pick the value computed for the previous template in the loop.
You should initialize the variable in the for loop.